### DISCUSSION TITLE
Feature Request: Custom Fields

### DISCUSSION THREAD

**Discussion by TobenderZephyr** [jonaswinkler/paperless-ng#1319] [Feature Requests]
I recently stumbled upon your project after trying other document management systems (like Papermerge).
First of all, thank you for taking your time to make this great project and update the old base!

I really like the functionality a lot because of so many importers which did not work well in other DMS (email in this case), but I was wondering, why certain other functions were missing: the ability to add certain custom fields (free text, currency, date, number). In Papermerge you were able to assign those fields onto folders where your documents reside in and so you were able to add more information to the documents and see everything in lists afterwards.

I do not use the field ASN at the moment, but it is always displayed in the views.

I would find it useful to add a currency field (price) to every receipt so you would see those information when searching for certain documents. Since you define your documents types beforehand (one of it being receipts), why not add custom fields to those types?

If this has been answered before, sorry for bringing it up again.

Kind regards,
Marcus

**Comment by TheFehr** [comment]
There has been work done regarding custom metadata over at #274.

Not sure it covers your wishes and sadly it has been not progressing for quite some time.

**Reply by TobenderZephyr** [reply]
Thank you! That is actually exactly what I was looking for. It seems once post consume hooks are available as well, it should be pretty straight forward to fill in those fields automatically as well. (Or so I hope).

I hope there are some updates to this somewhere.

 If I had any knowledge of angular/ python I would create a PR myself
